randovania/games/planets_zebeth/exporter/patch_data_factory.py

```diff
@@ -1,12 +1,136 @@
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
+from randovania.exporter import item_names, pickup_exporter
 from randovania.exporter.patch_data_factory import PatchDataFactory
+from randovania.game_description.assignment import PickupTarget
 from randovania.games.game import RandovaniaGame
+from randovania.generator.pickup_pool import pickup_creator
+from randovania.lib import json_lib
+
+if TYPE_CHECKING:
+    from random import Random
+
+    from randovania.exporter.pickup_exporter import ExportedPickupDetails
+    from randovania.games.planets_zebeth.layout.planets_zebeth_configuration import PlanetsZebethConfiguration
+    from randovania.games.planets_zebeth.layout.planets_zebeth_cosmetic_patches import PlanetsZebethCosmeticPatches
 
 
 class PlanetsZebethPatchDataFactory(PatchDataFactory):
+    cosmetic_patches: PlanetsZebethCosmeticPatches
+    configuration: PlanetsZebethConfiguration
+
+    def _create_pickups_dict(self, pickup_list: list[ExportedPickupDetails], item_info: dict, rng: Random) -> dict:
+        pickup_map_dict = {}
+        for pickup in pickup_list:
+            quantity = pickup.conditional_resources[0].resources[0][1] if not pickup.other_player else 0
+            object_id = self.game.region_list.node_from_pickup_index(pickup.index).extra["object_id"]
+            res_lock = pickup.original_pickup.resource_lock
+            text_index = (
+                1
+                if (
+                    res_lock is not None
+                    and "Locked" in res_lock.temporary_item.long_name
+                    and len(pickup.collection_text) > 1
+                )
+                else 0
+            )
+
+            pickup_map_dict[object_id] = {
+                "model": pickup.model.name,
```

Comment:
out of curiosity, hows the animation speed handled? do you just have a mapping in game with sprite name -> animation speed? Or do all models have the same speed?

Reply:
it's pretty much handled by the patcher. most of the pickups have no animations except for missile tanks.

Reply:
gotcha.